tools/RAiDER/utilFcns.py

```diff
@@ -157,25 +151,16 @@ def gdal_open(fname, returnProj=False, userNDV=None):
         return data, proj, gt
```

Comment:
It may be advantageous to instead just return the rasterio profile here when `returnProj == True`. That could potentially make code more general and subsequent writing of rasters more streamlined. The projection and geotransform can be accessed as keys in this profile dictionary e.g.

```
with rasterio.open(filename) as ds:
    profile = ds.profile
proj = profile['crs']
gt = profile['transform']
```

This is obviously potentially annoying due to change in return types. It will simplify code when *writing* after reading a similar raster e.g.

```
with rasterio.open(filename_0) as ds:
    X = ds.read()
    p = ds.profile

with rasterio.open(filename_1, 'w', **p) as ds:
   ds.write(X + 1)
```

Anyways, just an idea.

Reply:
+1 on this idea if it doesn't introduce too much complexity / other issues. I think this could actually simplify the code since all the attributes would get dragged along instead of just one or two .

Reply:
We seem to modify data type.  I will keep this as is for now - if we strongly feel about this; we can go in and update everything to use profile instead.